hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java

```diff
@@ -89,8 +92,8 @@ private void findNext() {
             throw new IllegalStateException("");
           }
 
-          if (!deletingTxIDs.contains(txID) &&
-              !skippingRetryTxIDs.contains(txID)) {
+          if (deletingTxIDs != null && !deletingTxIDs.contains(txID) &&
```

Comment:
```
 (deletingTxIDs == null || !deletingTxIDs.contains(txID)) &&
 (skippingRetryTxIDs == null || !skippingRetryTxIDs.contains(txID))
```

Reply:
Thanks for pointing out. Will address in the next patch.